libtelemetry/src/main/java/com/mapbox/services/android/telemetry/MapboxTelemetry.java

```diff
@@ -53,37 +53,38 @@
   private boolean isOpted = false;
   private boolean serviceBound = false;
   private PermissionCheckRunnable permissionCheckRunnable = null;
+  protected CopyOnWriteArrayList<TelemetryListener> telemetryListeners = new CopyOnWriteArrayList<TelemetryListener>();
 
-  public MapboxTelemetry(Context context, String accessToken, String userAgent, Callback httpCallback) {
+  public MapboxTelemetry(Context context, String accessToken, String userAgent) {
     if (checkRequiredParameters(accessToken, userAgent)) {
       initializeTelemetryClient();
     }
 
     this.context = context;
-    this.httpCallback = httpCallback;
     initializeQueue();
-    AlarmReceiver alarmReceiver = obtainAlarmReceiver(httpCallback);
+    AlarmReceiver alarmReceiver = obtainAlarmReceiver();
     this.schedulerFlusher = new SchedulerFlusherFactory(context, alarmReceiver).supply();
+    this.telemetryListeners.add(this);
   }
 
   // For testing only
   MapboxTelemetry(Context context, String accessToken, String userAgent, EventsQueue queue,
-                  TelemetryClient telemetryClient, Callback httpCallback, SchedulerFlusher schedulerFlusher,
-                  Clock clock, LocalBroadcastManager localBroadcastManager) {
+                  TelemetryClient telemetryClient, TelemetryListener telemetryListener,
```

Comment:
This should be tackled differently 👀 https://github.com/mapbox/mapbox-events-android/pull/38#discussion_r161520797